plasmapy/dispersion/numerical/tests/test_kinetic_alfven_.py

```diff
@@ -0,0 +1,179 @@
+        "kwargs, _error",
+        [
+            ({**_kwargs_single_valued, "B": "wrong type"}, TypeError),
+            ({**_kwargs_single_valued, "B": [1e-9, 2e-9, 3e-9] * u.T}, ValueError),
+            ({**_kwargs_single_valued, "B": -1 * u.T}, ValueError),
+            ({**_kwargs_single_valued, "B": 5 * u.m}, u.UnitTypeError),
+            ({**_kwargs_single_valued, "ion": "not a particle"}, InvalidParticleError),
+            ({**_kwargs_single_valued, "ion": "e-"}, ValueError),
+            ({**_kwargs_single_valued, "k": np.ones((3, 2)) * u.rad / u.m}, ValueError),
+            ({**_kwargs_single_valued, "k": 0 * u.rad / u.m}, ValueError),
+            ({**_kwargs_single_valued, "k": -1.0 * u.rad / u.m}, ValueError),
+            ({**_kwargs_single_valued, "k": 5 * u.s}, u.UnitTypeError),
+            ({**_kwargs_single_valued, "n_i": "wrong type"}, TypeError),
+            ({**_kwargs_single_valued, "n_i": [5e6, 6e6] * u.m**-3}, ValueError),
+            ({**_kwargs_single_valued, "n_i": -5e6 * u.m**-3}, ValueError),
+            ({**_kwargs_single_valued, "n_i": 2 * u.s}, u.UnitTypeError),
+            ({**_kwargs_single_valued, "T_e": "wrong type"}, TypeError),
+            ({**_kwargs_single_valued, "T_e": [1.4e6, 1.7e6] * u.K}, ValueError),
+            ({**_kwargs_single_valued, "T_e": -10 * u.eV}, ValueError),
+            ({**_kwargs_single_valued, "T_e": 2 * u.s}, u.UnitTypeError),
+            ({**_kwargs_single_valued, "T_i": "wrong type"}, TypeError),
+            ({**_kwargs_single_valued, "T_i": [4e5, 5e5] * u.K}, ValueError),
+            ({**_kwargs_single_valued, "T_i": -1 * u.eV}, ValueError),
+            ({**_kwargs_single_valued, "T_i": 2 * u.s}, u.UnitTypeError),
+            ({**_kwargs_single_valued, "theta": np.ones((3, 2)) * u.deg}, ValueError),
+            ({**_kwargs_single_valued, "theta": 5 * u.eV}, u.UnitTypeError),
+            ({**_kwargs_single_valued, "gamma_e": "wrong type"}, TypeError),
+            ({**_kwargs_single_valued, "gamma_i": "wrong type"}, TypeError),
+            ({**_kwargs_single_valued, "z_mean": "wrong type"}, TypeError),
+        ],
+    )
+    def test_raises(self, kwargs, _error):
+        """Test scenarios that raise an `Exception`."""
+        with pytest.raises(_error):
+            kinetic_alfven(**kwargs)
+
+    @pytest.mark.parametrize(
+        "kwargs, expected",
+        [
+            (
+                {
+                    **_kwargs_single_valued,
+                    "ion": Particle("He"),
+                    "z_mean": 2.0,
+                    "theta": 0 * u.deg,
+                },
+                {**_kwargs_single_valued, "ion": Particle("He +2"), "theta": 0 * u.deg},
+            ),
+            #
+            # z_mean defaults to 1
+            (
+                {**_kwargs_single_valued, "ion": Particle("He"), "theta": 0 * u.deg},
+                {**_kwargs_single_valued, "ion": Particle("He+"), "theta": 0 * u.deg},
+            ),
```

Comment:
```suggestion
            # The following test may need to be updated when applying
            # @particle_input to kinetic_alfven, since this refers to how
            # z_mean had been assumed to default to 1
            (
                {**_kwargs_single_valued, "ion": Particle("He"), "theta": 0 * u.deg},
                {**_kwargs_single_valued, "ion": Particle("He+"), "theta": 0 * u.deg},
            ),
```